PayloadPkg/OsLoader/OsLoader.c

```diff
@@ -668,13 +673,36 @@ StartBooting (
     MultiBoot = &LoadedImage->Image.MultiBoot;
     BeforeOSJump ("Jumping into FV/PE32 ...");
 
+    ZeroMem (&PldModParam, sizeof(PldModParam));
+    PldModParam.GetProcAddress = GetProcAddress;
+    OsBootOptionList = GetBootOptionList ();
+    PldModParam.CmdLineBuf =  "";
+
+    // For Setup payload module, pass the python script path as parameter.
+    if (!((OsBootOptionList == NULL) || (OsBootOptionList->OsBootOptionCount == 0))) {
+      CurrIdx = GetCurrentBootOption (OsBootOptionList, mCurrentBoot);
+      PathPtr = OsBootOptionList->OsBootOption[CurrIdx].Image[0].FileName;
+      if (AsciiStrnCmp ((CONST CHAR8 *)PathPtr, "!SETP/MPYM", 10) == 0) {
+        if (PathPtr[10] == ':') {
+          // !SETP/MPYM:????, here ???? is the script component name
+          // So build a path like !SETP/???? for command line.
+          CopyMem (ModCmdLineBuf, PathPtr, 6);
+          CopyMem (ModCmdLineBuf + 6, PathPtr + 11, 4);
+          ModCmdLineBuf[15] = 0;
+        } else {
+          ModCmdLineBuf[0] = 0;
+        }
+        PldModParam.CmdLineBuf = ModCmdLineBuf;
+      }
+    }

```

Comment:
Had better these code is built out by FeaturePcdGet (PcdEnableSetup) for size.

Reply:
We might need another PCD to control the payload module support.   This is different from the Setup support.   
Payload module will enable launching from OsLoader.   Setup support will control if we can edit CFGDATA. 
I can work on it.